server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1020,7 +1020,9 @@ private void writeUpdatedShardMetaDataAndComputeDeletes(
 
         // Listener that flattens out the delete results for each index
         final ActionListener<Collection<ShardSnapshotMetaDeleteResult>> deleteIndexMetadataListener = new GroupedActionListener<>(
-            onAllShardsCompleted.map(res -> res.stream().flatMap(Collection::stream).toList()),
+            onAllShardsCompleted.map(
+                res -> res.stream().filter(Objects::nonNull).flatMap(Collection::stream).filter(Objects::nonNull).toList()
```

Comment:
I think we can't have any `null` values in `res` can we, the grouped-action listener filters those out? How'd you reproduce/see an NPE here?

Reply:
Huh TIL. Sorry for the noise.